apps/src/studio/studio.js

```diff
@@ -554,6 +554,17 @@ function callHandler (name, allowQueueExtension) {
   });
 }
 
+/**
+ * Calls provided function using our api - i.e. treats func as if it's code
+ * generated from blocks.
+ */
+Studio.callApiCode = function (name, func) {
```

Comment:
Yeah, it's a little bit weird. Essentially it creates a command queue of the given name, and registers it with the provided function. It then calls the handler, which means the function gets executed, which fills the command queue. Finally, we execute the commands in the queue. Really, it's just a way of saying "execute the provided function in a way that behaves similarly to if it was code generated from blocks".